llama-index-core/tests/workflow/test_service.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from llama_index.core.workflow.decorators import step
+from llama_index.core.workflow.events import Event, StartEvent, StopEvent
+from llama_index.core.workflow.workflow import Workflow
+from llama_index.core.workflow.service import ServiceManager
+
+
+class ServiceWorkflow(Workflow):
+    """This wokflow is only responsible to generate a number, it knows nothing about the caller."""
+
+    @step()
+    async def generate(self, ev: StartEvent) -> StopEvent:
+        return StopEvent(result=42)
+
+
+class NumGenerated(Event):
+    """To be used in the dummy workflow below."""
+
+    num: int
+
+
+class DummyWorkflow(Workflow):
+    """
+    This workflow needs a number, and it calls another workflow to get one.
+    A service named "service_workflow" must be added to `DummyWorkflow` for
```

Comment:
is there some validation that we can implement to ensure that all necessary services have been registered?

Reply:
Good point. Yes, we can ensure that the service is at least registered and available through the service manager. Things can still go awry at runtime, but at least we can validate the system at rest, I'll add it.